tests/Client/UpdaterTest.php

```diff
@@ -314,6 +315,262 @@ public function providerVerifiedDelegatedDownload(): array
+                    'a' => 1,
+                    'b' => 1,
+                    'c' => null,
+                    'd' => null,
+                ],
+            ],
+            // Test using the TUFTestFixture3LevelDelegation fixture set.
+            'TUFTestFixture3LevelDelegation targets.txt' => [
+                'TUFTestFixture3LevelDelegation',
+                'targets.txt',
+                [
+                    'root' => 2,
+                    'timestamp' => 2,
+                    'snapshot' => 2,
+                    'targets' => 2,
+                    'a' => null,
+                    'b' => null,
+                    'c' => null,
+                    'd' => null,
+                    'e' => null,
+                    'f' => null,
+                ],
+            ],
+            'TUFTestFixture3LevelDelegation a.txt' => [
+                'TUFTestFixture3LevelDelegation',
+                'a.txt',
+                [
+                    'root' => 2,
+                    'timestamp' => 2,
+                    'snapshot' => 2,
+                    'targets' => 2,
+                    'a' => 1,
+                    'b' => null,
+                    'c' => null,
+                    'd' => null,
+                    'e' => null,
+                    'f' => null,
+                ],
+            ],
+            'TUFTestFixture3LevelDelegation b.txt' => [
+                'TUFTestFixture3LevelDelegation',
+                'b.txt',
+                [
+                    'root' => 2,
+                    'timestamp' => 2,
+                    'snapshot' => 2,
+                    'targets' => 2,
+                    'a' => 1,
+                    'b' => 1,
+                    'c' => null,
+                    'd' => null,
+                    'e' => null,
+                    'f' => null,
+                ],
+            ],
+            'TUFTestFixture3LevelDelegation c.txt' => [
+                'TUFTestFixture3LevelDelegation',
+                'c.txt',
+                [
+                    'root' => 2,
+                    'timestamp' => 2,
+                    'snapshot' => 2,
+                    'targets' => 2,
+                    'a' => 1,
+                    'b' => 1,
+                    'c' => 1,
+                    'd' => null,
+                    'e' => null,
+                    'f' => null,
+                ],
+            ],
+            'TUFTestFixture3LevelDelegation d.txt' => [
+                'TUFTestFixture3LevelDelegation',
+                'd.txt',
+                [
+                    'root' => 2,
+                    'timestamp' => 2,
+                    'snapshot' => 2,
+                    'targets' => 2,
+                    'a' => 1,
+                    'b' => 1,
+                    'c' => 1,
+                    'd' => 1,
+                    'e' => null,
+                    'f' => null,
+                ],
+            ],
+            'TUFTestFixture3LevelDelegation e.txt' => [
```

Comment:
It'd be nice if all these cases had some comments above them explaining the expected outcome.